lib/publiccloud/provider.pm

```diff
@@ -609,7 +609,7 @@ This method is called called after each test on failure or success.
 =cut
 sub cleanup {
     my ($self) = @_;
-    $self->terraform_destroy();
+    $self->terraform_destroy() unless get_var('PUBLIC_CLOUD_SERVICE') ~~ ["EKS"];
```

Comment:
why not?

```suggestion
    $self->terraform_destroy() unless check_var('PUBLIC_CLOUD_SERVICE', 'EKS');
```

Reply:
The idea is to have an array of excepted services. Is expected we'll have one for each public cloud.

Reply:
The idea is to have an array of excepted services. Is expected we'll have one for each public cloud.

Reply:
Ok, makes sense. Anyway, in a later PR we could improve the cleanup function based on @cfconrad idea: not running terraform destroy if terraform directory exists `script_run('test -d ' . TERRAFORM_DIR . '/.terraform');`

Reply:
Seems that pipeline complains about this. I haven't never seen this way of checking if "string is in array" in perl and I am not perl expert, but what I would probably do is:

`unless (grep $_ eq get_var('PUBLIC_CLOUD_SERVICE'), ('X', 'Y', 'Z')) {`

@cfconrad wdyt?

Reply:
I implemented the first solution and the VR is green.